src/Castle.Windsor/MicroKernel/Handlers/DefaultGenericHandler.cs

```diff
@@ -280,6 +280,12 @@ private void CloneParentProperties(ComponentModel newModel)
 			}
 		}
 
+		public IHandler ConvertToClosedGenericHandler(Type service, CreationContext openGenericContext)
+		{
+			var closedType = this.GetClosedImplementationType(openGenericContext, true); // Not sure about required
```

Comment:
`instanceRequired` seems to basically determine whether `DefaultGenericHandler` will throw (`true`) or return null (`false`). Since you are using this from `CanResolve` you probably want `false`.

Reply:
Thanks for this. It is very very important to note that this logic is now used from both can resolve and resolve paths. The fix however was really in the default sub resolver during the resolve process. It was a bastard and took a full day of debugging to work it out.